### PR TITLE
*: refine etcd transaction errors

### PR DESCRIPTION
**Pull request by JmPotato**
Signed-off-by: JmPotato <ghzpotato@gmail.com>

### What problem does this PR solve?

The old `ErrEtcdTxn` error is ambiguous to explain why the transaction failed.

### What is changed and how it works?

This PR distinguishes the two different errors of etcd internal failure and transaction conflict rollback.

### Check List

Tests

- Unit test
- Integration test

### CODE REVIEW

**Comment by HunDunDM** [comment]
@rleungx Will this introduce incompatible behavior?

**Review by Yisaer — APPROVED** [review]
LGTM

File: errors.toml [hunk 1]
```diff
@@ -261,9 +261,14 @@ error = '''
 etcd TLS config error
 '''
 
-["PD:etcd:ErrEtcdTxn"]
+["PD:etcd:ErrEtcdTxnConflict"]
 error = '''
-etcd Txn failed
+etcd transaction failed, conflicted and rollbacked
```
Comment (rleungx) [inline errors.toml:266]: rolled back?

**Review by rleungx — COMMENTED** [review]
The rest LGTM

File: errors.toml (hunk 1 above)
Comment (JmPotato) [reply errors.toml]: Fixed. PTAL.

**Review by rleungx — APPROVED** [review]

**Pull request merged**